metrics.go

```diff
@@ -43,6 +43,27 @@ const (
 	// MetricTrustedClusters counts trusted clusters
 	MetricTrustedClusters = "trusted_clusters"
 
+	// MetricClusterNameNotFound counts times a cluster name was not found
+	MetricClusterNameNotFound = "cluster_name_not_found_total"
+
+	// MetricFailedLoginAttempts counts failed login attempts
+	MetricFailedLoginAttempts = "failed_login_attempts_total"
+
+	// MetricFailedConnectToNodeAttempts counts failed ssh attempts
+	MetricFailedConnectToNodeAttempts = "failed_connect_to_node_attempts_total"
+
+	// MetricUserMaxConcurrentSessionsHit counts number of times the user exceeded their max concurrent ssh connections
+	MetricUserMaxConcurrentSessionsHit = "user_max_concurrent_sessions_hit_total"
+
+	// MetricProxyConnectionLimitHit counts the number of times proxy connection limit was exceeded
+	MetricProxyConnectionLimitHit = "proxy_connection_limit_exceeded_total"
+
+	// MetricUserLoginCount counts user logins
+	MetricUserLoginCount = "user_login_count"
+
+	// MetricHeartbeatConnectionsReceived counts heartbeat connections received by auth
```

Comment:
Not sure on the terminology but perhaps these would be better described as heartbeat messages? Correct me if I am wrong but I think a connection is persistently held and used for heartbeats.

Reply:
thanks for pointing this out because i did actually hook at the wrong place for heartbeat connections (in `grpcserver.go` )-- it should still be counting heartbeat connections but based on where i initally hooked -- i believe it counts messages. i will fix this.